tls/vibe/stream/openssl.d

```diff
@@ -1225,6 +1250,51 @@ final class OpenSSLContext : TLSContext {
 
 alias SSLState = ssl_st*;
 
+static if (haveKeylog) {
+	/**
+	 * Use the SSLKEYLOGFILE environment variable to log the key.
+	 *
+	 * Whatever filename is stored in the SSLKEYLOGFILE environment variable
+	 * determines where the keys will be logged. Each key will be logged on a
+	 * separate line.
+	 *
+	 * This must be called on every OpenSSLContext you wish to log the key. If the
+	 * environment variable is not set, this does not log the key.
+	 */
+	import vibe.core.file;
+	private FileStream keyfile;
+	void keylogOnEnvVar(OpenSSLContext context) {
+		static extern(C) void callback(const SSL* ssl, const char* line) {
+			if (line is null) return;
+
+			// use a separate boolean to avoid always checking for an env var when it doesn't exist.
+			static bool initialized;
+			if (!initialized) {
+				initialized = true;
+				// read the environment variable
+				import std.process;
+
+				import vibe.core.path;
+				auto path = NativePath(environment.get("SSLKEYLOGFILE", null));
+				if (path.empty) return;
+				// we use append like latest openssl does.
+				keyfile = openFile(path, FileMode.append);
+			}
+
+			if (!keyfile.isOpen) return;
+			keyfile.write(line[0 .. strlen(line)]);
+			keyfile.write("\n");
+		}
+
+		context.keylogCallback = &callback;
+	}
+
+	static ~this() {
+		keyfile.close();
+	}
```

Comment:
Are we sure that this can never be called from a different thread ? The context would have to leak, obviously.

Reply:
keyfile is TLS. so a TLS static destructor seems like the right place to close it.

Reply:
It's now a `__gshared File` from phobos. Vibe was too annoying to do this well.